packages/react-jss/src/JssProvider.js

```diff
@@ -18,65 +20,79 @@ type Props = {
   generateId?: GenerateId,
   classNamePrefix?: string,
   disableStylesGeneration?: boolean,
+  sheetOptions: StyleSheetFactoryOptions,
   children: Node
 }
 
-const defaultGenerateId = createGenerateIdDefault()
-
 export default class JssProvider extends Component<Props> {
   static propTypes = {
     registry: PropTypes.instanceOf(SheetsRegistry),
     jss: PropTypes.shape({}),
     generateId: PropTypes.func,
     classNamePrefix: PropTypes.string,
     disableStylesGeneration: PropTypes.bool,
-    children: PropTypes.node.isRequired
+    children: PropTypes.node.isRequired,
+    sheetOptions: PropTypes.shape({})
   }
 
+  managers: {} = {}
+
+  generateId = createGenerateId()
+
   createContext(outerContext: Context): Context {
-    const {registry, classNamePrefix, jss, generateId, disableStylesGeneration} = this.props
+    const {
+      registry,
+      classNamePrefix,
+      jss,
+      generateId,
+      disableStylesGeneration,
+      sheetOptions
+    } = this.props
     // Clone the outer context
-    const context = {...outerContext}
+    const context = {
+      ...outerContext,
+      managers: this.managers
+    }
 
-    if (registry) {
-      context.registry = registry
-      // This way we identify a new request on the server, because user will create
-      // a new Registry instance for each.
-      if (registry !== this.registry) {
-        // We reset managers because we have to regenerate all sheets for the new request.
-        this.managers = {}
-        this.registry = registry
+    if (sheetOptions) {
+      warning(
+        'generateId' in sheetOptions,
+        '[JssProvider] Do not pass the generateId inside the sheet options! We have a special prop for that!'
```

Comment:
I think we always used [jss] so far, I am not sure what the convention should be, but we should decide on that first. For e.g. it could be package name.

A better wording would be "Pass generateId as a prop instead of sheet options."

Reply:
I think we should actually stick with [JSS] just to let user know what project this warning is related to, they might not know the package name anyways

Reply:
👍